examples/text-generation/run_generation.py

```diff
@@ -293,19 +309,46 @@ def generate():
         HabanaProfile.disable()
         # Compilation
         logger.info("Graph compilation...")
+        dyn_prompt_lens = args.simulate_dyn_prompt
         t0 = time.perf_counter()
         # The first three iterations take longer because of graph compilation
-        for _ in range(args.warmup):
-            generate()
+        if dyn_prompt_lens is None or len(set(dyn_prompt_lens)) == 1:
+            for _ in range(args.warmup):
+                if dyn_prompt_lens is None:
+                    print("Warming up", flush=True)
+                    generate(None, args.reduce_recompile)
+                else:
+                    print("Warming up for shape,", dyn_prompt_lens[0], flush=True)
+                    generate(dyn_prompt_lens[0], args.reduce_recompile)
+        else:
+            if args.bucket_size > 0:
+                mn = min(dyn_prompt_lens)
+                mx = max(dyn_prompt_lens)
+
+                def rounder(x):
+                    return int(math.ceil(x / args.bucket_size) * args.bucket_size)
+
+                min_prompt_len = rounder(mn)
+                max_sentence_len = rounder(mx)
+                for _ in range(args.warmup):
+                    lst = list(range(min_prompt_len, max_sentence_len + 1, args.bucket_size))
+                    for sz in lst:
+                        print("Warming up for shape,", sz - 1, flush=True)
+                        generate(sz - 1, args.reduce_recompile)
         torch_hpu.synchronize()
         compilation_duration = time.perf_counter() - t0
         HabanaProfile.enable()
         total_new_tokens_generated = 0
         logger.info("Running generate...")
         t0 = time.perf_counter()
         # Benchmark over n_iterations iterations
-        for i in range(args.n_iterations):
-            generated = generate()
+        if dyn_prompt_lens is None:
+            for i in range(args.n_iterations):
+                generated = generate(None, args.reduce_recompile)
+        else:
+            for i in range(args.n_iterations):
+                print("Generating for shape,", dyn_prompt_lens[i])
```

Comment:
This will raise an error if `args.n_iterations` is larger than the length of `dyn_prompt_lens`

Reply:
fixed